core-plugins/src/main/java/io/cdap/plugin/transform/JavaScriptTransform.java

```diff
@@ -271,6 +273,22 @@ private InvalidEntry<StructuredRecord> getErrorObject(Map result, StructuredReco
   private Object decode(Object object, Schema schema) {
     Schema.Type type = schema.getType();
 
+    Schema.LogicalType logicalType = schema.getLogicalType();
+    if (logicalType != null) {
+      switch (logicalType) {
+        case DECIMAL:
+          BigDecimal bigDecimal = null;
+          if (object instanceof Double) {
+            bigDecimal = BigDecimal.valueOf((Double) object).setScale(schema.getScale(), RoundingMode.HALF_DOWN);
```

Comment:
why `HALF_DOWN` and not `HALF_EVEN`?

Reply:
This is an open question in the design, i have no preference here !
Let me know what works the best !

Reply:
I think `HALF_EVEN` is the most commonly used one in applications where fairness is important, as it tends to distribute rounding errors more evenly across a large number of calculations.

Reply:
Updated to `HALF_EVEN` !